marimbabot_planning/scripts/self_calibration_optimization_example.ipynb

```diff
@@ -0,0 +1,355 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Plot a given marimba model as seen from the world frame\n",
+    "def plot_model(model, colors=None):\n",
+    "    # Handle default case\n",
+    "    if colors is None:\n",
+    "        colors = np.random.rand(len(model), 3)\n",
+    "\n",
+    "    fig = plt.figure()\n",
+    "    ax = fig.add_subplot(111)\n",
+    "\n",
+    "    # Add bars to plot\n",
+    "    for i, bar in enumerate(model):\n",
+    "        xs = bar[:,0]\n",
+    "        ys = bar[:,1]\n",
+    "        ax.plot(xs, ys, c=colors[i])\n",
+    "\n",
+    "    # Set the axis labels and title\n",
+    "    ax.set_xlabel('X')\n",
+    "    ax.set_ylabel('Y')\n",
+    "    ax.set_title('Marimba Model')\n",
+    "    \n",
+    "    # Set the aspect ratio to 'equal' for all axes\n",
+    "    ax.set_aspect('equal')\n",
+    "\n",
+    "model = generate_marimba_model(12, 3, 1)\n",
+    "plot_model(model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 483,
+   "id": "63d49f0e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Creates a transformation matrix based on a rotation (radians) and translation\n",
+    "def get_transformation(rotation_angle, translation):\n",
+    "    cos_theta = np.cos(rotation_angle)\n",
+    "    sin_theta = np.sin(rotation_angle)\n",
+    "    return np.array([[cos_theta, -sin_theta, translation[0]],\n",
+    "                  [sin_theta, cos_theta, translation[1]],\n",
+    "                  [0, 0, 1]])\n",
+    "\n",
+    "# Transforms the bars of the marimba model according to a given transformation matrix in a vectorized way\n",
+    "def transform_model(lines, T):\n",
+    "    # Extract the rotation and translation components from the transformation matrix\n",
+    "    R = T[:2, :2]\n",
+    "    t = np.hstack((T[:2, 2], [0]))\n",
+    "    \n",
+    "    # Apply the rotation to the model\n",
+    "    rotated_model = np.einsum('ijk,lk->ijl', lines, T)\n",
```

Comment:
what does the argument mean?

Reply:
https://en.wikipedia.org/wiki/Einstein_notation

Reply:
It is a short notation for what is essentially a batched matrix multiplication.